src/debug/injected/recorder.ts

```diff
@@ -92,24 +92,69 @@ export class Recorder {
     });
   }
 
-  private _buildSelector(node: Node): string {
-    const element = node as Element;
-    for (const attribute of ['data-testid', 'aria-label', 'id', 'data-test-id', 'data-test']) {
+  private _buildSelector(targetElement: Element): string {
+    const path: string[] = [];
+    const root = document.documentElement;
+    for (let element: Element | null = targetElement; element && element !== root; element = element.parentElement) {
+      const selector = this._buildSelectorCandidate(element);
+      if (selector)
+        path.unshift(selector.selector);
+
+      const fullSelector = path.join(' ');
+      if (selector && selector.final)
```

Comment:
I am curious why don't you do the sanity check for "final" selectors?

Reply:
Because it is wip!